### PR TITLE
unix_sigpipe: Add test for SIGPIPE disposition in child processes

### PR DESCRIPTION
**Pull request by Enselic**
To make it clearer what the impact would be to stop using `SIG_IGN` and instead use a noop handler, like suggested [here](https://github.com/rust-lang/rust/issues/62569#issuecomment-1961586025) and implemented [here](https://github.com/rust-lang/rust/pull/121578).

Part of https://github.com/rust-lang/rust/issues/97889

### CODE REVIEW

File: tests/run-make/sigpipe-in-child-processes/main.rs
Comment (deltragon) [inline tests/run-make/sigpipe-in-child-processes/main.rs:13]:
```suggestion
#[cfg_attr(sig_ign, unix_sigpipe = "sig_ign")]
#[cfg_attr(inherit, unix_sigpipe = "inherit")]
```
Isn't this missing? I assume the test still works because it happens to be the same as the default.

Comment (Enselic) [reply tests/run-make/sigpipe-in-child-processes/main.rs:16]: Good catch, thanks! It indeed passed by accident. I extended the test in a way that would have caught this mistake.

**Comment by Mark-Simulacrum** [comment]
Is there a reason this is a run-make test? It seems like it should be just a run-pass test like this one https://github.com/rust-lang/rust/blob/master/tests/ui/panics/abort-on-panic.rs#L1-L3 (or plenty of others with run-pass and revisions:)

**Comment by Enselic** [comment]
@Mark-Simulacrum I don't think run-pass tests supports building the separate program that this test [needs](https://github.com/rust-lang/rust/pull/121573/files#diff-108eeb4adc14b1d5df9782a9d50cffdda2a7b6bbe8adc19651b81cc2cbcd4055)? The example you linked to runs itself as a child process, but my test can't do that, since the Rust runtime would mess with `SIGPIPE` in the child.

**Comment by Mark-Simulacrum** [comment]
Ah. I think it might be possible with auxiliary test files, but it's probably not worth the extra work then.

@bors r+

**Comment by jhpratt** [comment]
Failed in #122327: https://github.com/rust-lang/rust/pull/122327#issuecomment-1987798553

@bors r-

**Comment by Enselic** [comment]
@Mark-Simulacrum With one small but ugly hack I was able to turn the test into a run-pass test instead of a run-make test. 

The `aarch64` CI failure should also be fixed now by using `libc::c_char` instead of `i8`.

**Comment by Enselic** [comment]
I will split this up into smaller PRs for easier review (first out is https://github.com/rust-lang/rust/pull/122419).

**Comment by Enselic** [comment]
@Mark-Simulacrum After rebasing on https://github.com/rust-lang/rust/pull/122634 the only commit left in this PR is the test itself (which is now a regular run-pass ui test) which you more or less already approved, so would be great if you could take another look.

File: tests/ui/attributes/unix_sigpipe/unix_sigpipe-and-child-processes.rs
Comment (Mark-Simulacrum) [inline tests/ui/attributes/unix_sigpipe/unix_sigpipe-and-child-processes.rs:2]: This isn't present in most run-pass tests, so what makes this one special?

**Review by Mark-Simulacrum — CHANGES_REQUESTED** [review]
r=me if we really need the extra ignore

File: tests/ui/attributes/unix_sigpipe/unix_sigpipe-and-child-processes.rs
Comment (Enselic) [reply tests/ui/attributes/unix_sigpipe/unix_sigpipe-and-child-processes.rs]: Turns out it is not `run-pass` that is the problem here, because the following `run-pass` and `aux-build` test passes in cross-compilation context:

```console
src/ci/docker/run.sh --dev armhf-gnu
../x test --target arm-unknown-linux-gnueabihf tests/ui/cross-crate/reexported-static-methods-cross-crate.rs
```

Instead, it is `aux-bin` that does not support cross compilation yet. For example, `remote-test-client` does not yet know how to upload auxiliary binaries to the target device. That it something that would be nice to solve, but it does not block this PR. I updated the motivation for the ignore.

I would be happy to `@bors r=Mark-Simulacrum` but you'll need to `@bors delegate=Enselic` me first.

Comment (Mark-Simulacrum) [reply tests/ui/attributes/unix_sigpipe/unix_sigpipe-and-child-processes.rs]: Sounds good, thanks.

**Pull request merged**
